acct/chsh

```diff
@@ -6,32 +6,39 @@ import sys
 from ocflib.account.manage import modify_ldap_attributes
 from ocflib.misc.validators import VALID_LOGIN_SHELLS
 
-if len(sys.argv) < 2:
-    print('Usage: chsh [shell]')
-    sys.exit(1)
-
-new_shell = sys.argv[1]
-
-print('Changing shell to {}...'.format(new_shell))
-
-# Check if the user has a valid kerberos ticket
-has_kerberos = subprocess.run(('/usr/bin/klist', '-t')).returncode
-
-if has_kerberos != 0:
-    # The user doesn't have a valid kerberos ticket already, so obtain a new
-    # ticket (this asks for a password if they don't already have a ticket, and
-    # then uses their password to obtain a new ticket)
-    subprocess.run('/usr/bin/kinit', check=True)
-
-# modify_ldap_attributes already does loginShell validation for us to make sure
-# it's a valid shell, so we don't have to do any extra here
-try:
-    modify_ldap_attributes(getpass.getuser(), {'loginShell': new_shell})
-except ValueError as e:
-    print(e, 'valid shells are: {}'.format(list(VALID_LOGIN_SHELLS)))
-    sys.exit(1)
-
-print(
-    'Shell successfully changed, typically it takes a while for the '
-    'changes to propagate to all our machines'
-)
+
+def main(args):
+    if len(args) < 2:
+        print('Usage: chsh [shell]')
```

Comment:
The `shell` argument is actually mandatory.

I guess this is a topic for a separate PR, but can we change this script to conform more closely to the actual `chsh`? (E.g. use `-s` to specify the shell, if no shell specified then prompt for one)

Reply:
Good catch, it should have been marked mandatory.

As for the interface, I think that would probably be better for users to use the more standard interface so that it's easier to select a shell (interactively), but I doubt it's a big deal. Maybe if some scripts or programs expect to use the standard interface for `chsh` or something, but even that seems like a pretty obscure use case. Ultimately, since we've had `chsh` work like this for ages and haven't had any complaints about it that I know of, it doesn't seem like changing it would really have much impact either way.

Reply:
I remember being confused the first time around having chsh not have the semantics I expected from other machines because it was custom @ OCF. If we're going to override defaults like we are doing here it's probably a good idea to at least retain the interface.

Reply:
Ok, I've done this in the latest commit, I agree that making our `chsh` as standard as possible (while still editing the `loginShell` in LDAP) is a good move.